### ISSUE TITLE
Error initializing storage of type postgresql for PostgreSQL 10.2

### ISSUE THREAD

**Issue by vitya-ne** [hashicorp/vault#3949]
Vault initialization have an error after upgrade Postgresql to 10.2

This place in **postgresql.go** now wrong: 
```
var upsert_required bool
upsert_required_query := "SELECT string_to_array(setting, '.')::int[] < '{9,5}' FROM pg_settings WHERE name = 'server_version'"
```

on PostgreSQL 10.1:
```
SELECT string_to_array(setting, '.') FROM pg_settings WHERE name = 'server_version';
 string_to_array 
-----------------
 {10,1}

```
on PostgreSQL 10.2:
```
SELECT string_to_array(setting, '.') FROM pg_settings WHERE name = 'server_version';
          string_to_array          
-----------------------------------
 {10,"2 (Debian 10",2-1,pgdg90+1)}

```

* Vault Version: 0.9.3
* Operating System/Architecture: Debian 9.3

**Startup Log Output:**
`Error initializing storage of type postgresql: failed to check for native upsert: pq: неверное значение для целого числа: "2 (Debian 10"`

**Comment by jefferai** [comment]
It's unclear whether this is something Debian-specific or not; regardless this is a community supported backend so definitely the fastest way to get this resolved would be to put in a PR.

**Comment by vitya-ne** [comment]
@jefferai, thanks!
issue resolved in version 0.9.4.
